ARMmbed/theia
# File dialog fixes

#### What it does
- Fixes two issues with file dialog filter behaviour by cherry-picking two upstream the commits:
  - Custom filters aren't selected by default
  - Custom filters aren't applied unless re-selected

#### How to test
- Open a file dialog with a custom filter and ensure that:
   - The custom filter is selected by default
   - And the custom filter is correctly applied (only matching files are shown)

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
